exonum-java-binding/core/rust/src/testkit/mod.rs

```diff
@@ -242,38 +259,41 @@ fn instance_configs_from_java_array(
                 .get_field(artifact_spec_obj, "serviceSpecs", SERVICE_SPECS_FIELD_TYPE)?
                 .l()?
                 .into_inner();
-            // TODO: Avoid deploy arguments duplication after ECR-3690
-            let configs = parse_service_specs(env, service_specs_obj, artifact_id, deploy_args)?;
-            instance_configs.extend(configs);
+            let instance_params = parse_service_specs(env, service_specs_obj, artifact_id.clone())?;
+            let testkit_service = TestKitService {
+                artifact_id,
+                deploy_args,
+                instances: instance_params,
+            };
+            services.push(testkit_service);
 
             Ok(JObject::null())
         })?;
     }
-    Ok(instance_configs)
+    Ok(services)
 }
 
-// Converts Java array of `ServiceSpec` instances into vector of `InstanceConfig` for specific artifact.
+// Converts Java array of `ServiceSpec` instances into vector of `InstanceInitParams` for specific artifact.
 fn parse_service_specs(
     env: &JNIEnv,
     specs_array: jobjectArray,
-    artifact_id: String,
-    deploy_args: Vec<u8>,
-) -> JniResult<Vec<InstanceConfig>> {
+    artifact_id: ArtifactId,
+) -> JniResult<Vec<InstanceInitParams>> {
     let num_specs = env.get_array_length(specs_array)?;
 
-    let mut instance_configs = vec![];
+    let mut instances = vec![];
     for i in 0..num_specs {
         env.with_local_frame(8, || {
             let service_spec = env.get_object_array_element(specs_array, i)?;
-            let (spec, config) = parse_instance_spec(&env, service_spec, &artifact_id)?;
-            let cfg = InstanceConfig::new(spec, Some(deploy_args.to_bytes()), config);
-            instance_configs.push(cfg);
+            let (spec, constructor) = parse_instance_spec(&env, service_spec, artifact_id.clone())?;
+            let params = InstanceInitParams::new(spec.id, spec.name, spec.artifact, constructor);
```

Comment:
Wouldn't it be easier to construct InstanceInitParams in parse_instance_spec instead of passing here a tuple?